gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -223,7 +226,13 @@ private Object doResumableUpload() throws IOException {
 
       private final long writeOffset;
       private final String uploadId;
+      // A map holding cached data chunks, keyed by the offset to start writing each data chunk.
+      private final TreeMap<Long, ByteString> dataChunkMap;
```

Comment:
Why do we need 2 `dataChunkMap`s?

Reply:
+1, I started a previous conversation about this.
I suggest we set up a meeting because we seem to be revisiting the same questions.

Reply:
Discussed offline, removed the retryMap for now.